csbiginteger/BigInteger.py

```diff
@@ -67,56 +63,135 @@
+               bytes are parsed assuming little endian.
+            base: only used when converting from `str` type
+        """
         # 256 bytes, standard size (TODO: improve this with better logic, but now it's enough)
         self.std_size = 256
+        if type(param) is float:
+            param = int(param)
         if type(param) is int:
             param = str(param)  # convert to base-10 integer
             base = 10  # force base 10
         if type(param) is bytearray:
             param = bytes(param)  # bytearray to bytes
         if type(param) is bytes:
             self._length = len(param)
-            # size is not standard, is less. exactly what is needed
-            self._datasize = self._length
-            self._data = (ctypes.c_uint8 *
-                          self._datasize).from_buffer(bytearray(param))
+            self._data = (ctypes.c_uint8 * self._length).from_buffer(bytearray(param))
         if type(param) is str:
-            self._datasize = self.std_size
-            self._data = (ctypes.c_uint8*self._datasize)()
+            # allocate a standard buffer size to use in the library
+            self._data = (ctypes.c_uint8 * self.std_size)()
+
             strsize = len(param)
-            strdata = (ctypes.c_char *
-                       strsize).from_buffer(bytearray(param, 'ascii'))
-            sz = csbiginteger_lib.csbiginteger_init_s(
-                strdata, base, self._data, self._datasize)
+            strdata = (ctypes.c_char * strsize).from_buffer(bytearray(param, 'ascii'))
+
+            sz = csbiginteger_lib.csbiginteger_init_s(strdata, base, self._data, self.std_size)
             if sz == 0:
                 raise ValueError('Something wrong with BigInteger. Zero size.')
             self._length = sz
 
+            # trim data to exact length
+            alloc = (ctypes.c_uint8 * self._length)
+            new_data = alloc.from_buffer(bytearray(self._data)[:self._length])
+            self._data = new_data
+
         # more options here?
 
-    # returns value in signed int32 limit (or exception)
-    def to_int(self):
+    def __int__(self) -> int:
+        """
+        Convert value to `int` type.
+
+        Raises:
+            OverflowError: if value does not fit in an `int` according to the C# rules
+
+        Returns:
+             int: value in signed int32 limit
+        """
         if self.__gt__(2**31-1):
             raise OverflowError("overflow on signed int32")
         if self.__lt__(-2**31):
             raise OverflowError("underflow on signed int32")
         return int(str(self))
 
-    def to_long(self):
+    def __float__(self) -> float:
+        """
+        Convert value to `long` type.
+
+        Raises:
+            OverflowError: if value does not fit in a `long` according to the C# rules
+
+        Returns:
+             long: value in signed int32 limit
```

Comment:
This one is strange to... in a __float__ context... mentions long, and also a int32...